kroxylicious-kms-test-support/src/main/java/io/kroxylicious/kms/service/AbstractTestKekManager.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.service;
+
+import java.util.Objects;
+import java.util.concurrent.CompletionException;
+
+/**
+ * The type Abstract test kek manager.
+ */
+public abstract class AbstractTestKekManager implements TestKekManager {
+
+    @Override
+    public void generateKek(String alias) {
+        Objects.requireNonNull(alias);
+
+        if (exists(alias)) {
```

Comment:
From a very quick look, I think this PR goes in the wrong way.     The Vault and AWS TestKekManager use the "check then act" pattern.  That's an anti-pattern that should eliminated.   We should be relying on the KMS itself to detect conditions like the non-existence of KEKs.    We kinda touched on the issue here: https://github.com/kroxylicious/kroxylicious/issues/1376

In this PR, you are actually baking in the anti-pattern.  That what I mean when I say it goes in the wrong direction.
The InMemoryTestKekManager is the only KMS where the "check then act" is justifiable.

Reply:
Yes, I'm aware of that. I wanted to make this refactor first to avoid having many `generateKEK` methods and then modify the "check and act" in the issue you pointed. I didn't want to make both issues at the same time.

Is that okay for all of you? or do you prefer to go in this PR?

Reply:
> Is that okay for all of you? or do you prefer to go in this PR?

no, not really.   I think it would make sense to refactor the Vault and AWS implementations to remove the check-then-act, first.  Once that is done, I expect the motivation for the shared base class AbstractTestKekManager will disappear because there will be no duplicated code.